runtime-common/stdlib/string/string-context.h

```diff
@@ -30,13 +32,16 @@ inline constexpr std::string_view PERCENT_ = "%";
 
 }; // namespace string_context_impl_
 
-class StringLibContext final : vk::not_copyable {
+struct StringLibContext final : vk::not_copyable {
+  static constexpr int32_t STATIC_BUFFER_LENGTH = 1U << 23U;
+
+private:
   static constexpr int32_t MASK_BUFFER_LENGTH = 256;
 
-public:
-  static constexpr int32_t STATIC_BUFFER_LENGTH = 1U << 23U;
+  std::aligned_storage<STATIC_BUFFER_LENGTH + 1> storage;
 
-  std::array<char, STATIC_BUFFER_LENGTH + 1> static_buf{};
+public:
+  char *static_buf{reinterpret_cast<char *>(std::addressof(storage))};
```

Comment:
I think it's better to have a method `static_buf()`.
We would get rid of extra invariant for self-referential data structure so there will be a narrower space for bugs.

Reply:
Let's do it in a separate PR